### ISSUE TITLE
Accordion: Missing role on `.accordion-collapse` ?

### ISSUE THREAD

**Issue by louismaximepiton** [twbs/bootstrap#37591]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

I don't really what's the best here, but I think we should either add a `role="region"` or remove our `aria-labelledby="*"` from our `.accordion-collapse`.

### Motivation and context

There is one reason that made me think of this, and several things that make me struggle on the solution to choose, I'll try to present them here.

### Reason

Some observations on aXe DevTools inside my browser. It raises this:
![image](https://user-images.githubusercontent.com/91960143/205854506-a23481c1-541e-4704-ab24-9253d893b9ab.png)

### Add `role="region"`

I must admit that my heart is going to this solution since W3C seems to recommand it in their example page ([here](https://www.w3.org/WAI/ARIA/apg/example-index/accordion/accordion.html)). 

That said, it seems that several front-end framework/design system encourage to use this solution everytime such as [Material Angular](https://material.angular.io/cdk/accordion/overview) and [Adobe](https://opensource.adobe.com/spectrum-css/accordion.html).

As an alternative maybe we could use a `<section>` as well as in [Aditus](https://www.aditus.io/patterns/accordion/).

### Remove `aria-labelledby="*"`

On the other side, I could find several sources on web that doesn't implement the `role="region"`. I feel like it's more personal projects, but there are also some front-end frameworks that display it such as [Getuikit](https://getuikit.com/docs/accordion).

If we remove this one, maybe we should add a paragraph to mention that this role depends on the content whether it's significant or not ?
